### ISSUE TITLE
AWS Lambda (and / or other cloud vendor) example(s) of Connexion 3

### ISSUE THREAD

**Issue by Lewiscowles1986** [spec-first/connexion#1936]
### Description

So since October 2023 we've had a desire to upgrade from connexion 2 to connexion 3 in AWS Lambda using the FlaskApp

I've opened another thread about issues encountered https://github.com/spec-first/connexion/issues/1807

I Also have an example of the connexion quick-start in AWS Lambda

### Expected behaviour

Easy setup, well-trodden path

### Actual behaviour

Errors from libraries used within connexion (vendor dependencies)

### Steps to reproduce

Minimal reproducible example 
[Archive 3.zip](https://github.com/spec-first/connexion/files/15479776/Archive.3.zip)
Create a lambda, python 3.12 on ARM64; upload the zip

Building / iterating

```
pip install \
--platform manylinux2014_aarch64 \
--target=package \
--implementation cp \
--python-version 3.12 \
--only-binary=:all: --upgrade \
-r requirements.txt
```

**requirements.txt:**

```
connexion[swagger-ui,flask]==3.1.0
asgi_aws==2.0.0
```

### Additional info:

Output of the commands:

- `python --version` 3.12
- `pip show connexion | grep "^Version\:"` 3.1.0

https://github.com/abersheeran/a2wsgi/issues/56
https://github.com/yezz123/asgi-aws/issues/25

**Note:** So one of the issues above; I think it's a2wsgi a connexion dependency is patched in the zip I've attached to overcome an issue with the body returning empty. I still need to experiment more to find if that is necesarry, rather than plain convenient.

The only reason I've not switched to FastAPI, which was very painless to setup is that with connexion, I author the OpenAPI and then the program fails if it does not comply; wherease with FastAPI; it seems the default is inverted. Low developer friction at the high-high cost of not knowing when things break for FastAPI users.

This project (Connexion) is amazing; I'm open to working harder on this problem, being course corrected with simpler approach and contributing bug fixes or docs to connexion or upstream repos.

Thank you

**Comment by Lewiscowles1986** [comment]
Looks like mangum has a better ASGI lifecycle for managing Connexion apps. Just replaced asgi-aws with mangum, and things "Just work ™️"

I wonder if the team would be open to receiving an example or write-up?

**Comment by johnbyrne7** [comment]
AWS Lambda has a one-process-per-request model.  Will ASGI be all that helpful in AWS Lambda?  Correct me if wrong, but is not the big win for ASGI the ability to process many more requests more efficiently  with fewer process and threads?  Does this apply to Lambda?  Might be interesting to test

Of course, if your per-request processing would benefit from async coding, then do that, in any case.

**Comment by Lewiscowles1986** [comment]
yeah @johnbyrne7 what we've found (we're moving some workloads off lambda to ECS) is that async just helps to be able to coordinate things like database connections earlier and be able to reason about things happening together rather than purely sequentially; it might not be for everyone and on smaler lambdas, it might be slower (we got a 200ms slowdown recently in a workload); so in those cases we just switch up affected code; there are no silver bullets here.

**Comment by Lewiscowles1986** [comment]
As nobody has indicated a desire I'll close this. It wasnt' that hard once I stopped to think a bit.
